core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -39,225 +44,251 @@
  * The syntax of the metrics configuration file is:
  * [instance].[sink|source].[name].[options]=[value]
  */
-@NotThreadSafe
-public class MetricsSystem {
+@ThreadSafe
+public final class MetricsSystem {
```

Comment:
Discussed in person.

I think one of the reasons of moving away from singleton is for testing. But for metrics, when one is added in a class, usually we don't need to worry about testing. So I think it is fine. 

@aaudiber to take a look as well.